addons/html_editor/static/src/components/history_dialog/history_dialog.scss

```diff
@@ -1,168 +1,276 @@
 .html-history-dialog-container {
-    margin-left: 10px;
-    margin-right: 10px;
-    width: calc(100% - 20px);
-}
-.html-history-dialog {
-    position: relative;
-    .history-view-top-bar {
-        display: flex;
-        align-items: center;
-        padding: 8px 12px;
-        background-color: #f7f7f7;
-        border-bottom: 1px solid #ddd;
-
-        >div {
-            flex-grow: 6;
-            &.toggle-view-btns {
-                flex-grow: 1;
-                padding-right: 10px;
-                width: 220px;
-            }
-            &:last-child {
-                flex-grow: 1;
-                text-align: right;
-                width: 180px;
-                .fa {
-                    margin-right: 10px;
-                }
-            }
+    overflow: hidden;
+
+    .o_expand_button {
+        padding: calc(var(--modal-header-padding-y) * .5) calc(var(--modal-header-padding-x) * .5);
+        margin: calc(-.5 * var(--modal-header-padding-y)) var(--modal-header-padding-x) calc(-.5 * var(--modal-header-padding-y)) auto;
+        color: var(--btn-close-color);
+        background: transparent var(--btn-close-bg) center / 1em auto no-repeat;
+        border: 0;
+        border-radius: $border-radius-sm;
+        opacity: var(--btn-close-opacity);
+        width: 2rem;
+
+        &.fa-expand::before {
+            content: "\f065";
         }
-    }
-    .history-view-inner {
-        padding: 8px 12px;
-        border: 1px solid #ddd;
-        border-top: 0;
-        overflow: auto;
-        .embedded-history-dialog-placeholder {
-            color: #444;
-            padding: 32px;
-            text-align: center;
-            font-size: 20px;
-            border : 1px solid #999;
-            border-radius: 4px;
-            margin: 8px 0;
-            background-image: linear-gradient(45deg, #d1d1d1 25%, #999 25%, #999 50%, #d1d1d1 50%, #d1d1d1 75%, #999 75%, #999);
-            background-size: 50px 50px;
-            text-shadow:
-                    -2px -2px 0 #d1d1d1,
-                    2px -2px 0 #d1d1d1,
-                    -2px 2px 0 #d1d1d1,
-                    2px 2px 0 #d1d1d1,
-                    -3px 0px 0 #d1d1d1,
-                    3px 0px 0 #d1d1d1,
-                    0px -3px 0 #d1d1d1,
-                    0px 3px 0 #d1d1d1,
-        }
-        .history-comparison-split {
-            position: relative;
+
+        &:hover {
+            opacity: 0.75;
+        }
+
+        &:focus {
+            outline: 0;
+            box-shadow: $btn-close-focus-shadow;
+            opacity: 1;
+        }
+
+        &:active {
+            border: none;
+            transform: none;
         }
+
+        &:disabled,
+        &.disabled {
+            pointer-events: none;
+            user-select: none;
+            opacity: 0.25;
+        }
```

Comment:
In which cases is this button disabled? If this style is necessary, the opacity property won´t be applied (utilities overriding it).

For the opacity not applied, check other occurences (eg focus state)